### ISSUE TITLE
Memory overflow on vector comparison.

### ISSUE THREAD

**Issue by TDW89** [KSP-KOS/KOS#1520]
the code:
```
set vec1 to v(0,1,0).
set vec2 to v(0,1,0).
print vec1 = vec2.
```
causes the game to exit.
There is no information in the log files however starting the game from the terminal gives the message `Segmentation fault (core dumped)` as the game exits.

other structures tested (without crash):
 - list
 - direction
 - vessel
 - part
 - orbit
 - geocoordinates
 - lexicon
 - iterator 
 - rgb
 - node
 - body
 - CrewMember

**Comment by Dunbaratu** [comment]
Hmm.  Anything that crashes it is bad and that needs fixing, but on the other hand, should this even *be* supported in its current form at all?  Asking "is vector 1 equal to vector 2?" is exactly like asking "is float A equal to float B"?  It's the sort of question that really needs an epsilon qualifier, to transform it into "is A *close enough* to B that I'll consider them equal?"  Vectors are made of floating point numbers, and testing directly for rigid equality is usually a bad idea.

Then again, there are cases where it could be reliable if the values are known to be nice easy small rounded numbers, just like with doubles.  With a double, `if (x == 1.0) ... ` works just fine.  So I suppose we can support it.  I just don't think it's going to be used all that much.

**Comment by TDW89** [comment]
yep I wasn't expecting it to work, just that it would save a couple of lines for me if it did.
The issue was that the game window disappeared and didn't leave anything in the logs.

**Comment by hvacengi** [comment]
If I recall correctly, float equality offers an option with a tolerance.  In my initial fix I didn't bother looking for that, but we can certainly incorporate it if desired.

**Comment by hvacengi** [comment]
Fixed the recursion itself via 8a58855.  We will still be looking at doing some more things to improve performance and resilience of the calculators, but this issue itself can be closed.
